src/main/groovy/wooga/gradle/build/unity/ios/IOSBuildPlugin.groovy

```diff
@@ -196,6 +211,22 @@ class IOSBuildPlugin implements Plugin<Project> {
             it.xcarchivePath xcodeArchive
         }
 
+        def publishTestFlight = tasks.create(maybeBaseName(baseName, "publishTestFlight"), PublishTestFlight) {
+            it.ipa xcodeExport
+            it.group = PublishingPlugin.PUBLISH_TASK_GROUP
+            it.description = "Upload binary to TestFlightApp"
+        }
+
+        project.afterEvaluate(new Action<Project>() {
+            @Override
+            void execute(Project _) {
+                if(extension.publishToTestFlight) {
+                    def lifecyclePublishTask = tasks.getByName(PublishingPlugin.PUBLISH_LIFECYCLE_TASK_NAME)
+                    lifecyclePublishTask.dependsOn(publishTestFlight)
+                }
+            }
+        })
+
```

Comment:
Neat. Is AppCenter plugin doing the same, or it's a "new" approach you can use because you're in the same plugin and can hook up to xcodeExport?

Reply:
I have the patch created but not yet published as a PR for the appcenter plugin. I tend not to use `afterEvaluate` but in this case it is the easiest solution.
And yes the idea is to have a switch there as well.